quesma/quesma/config/config_v2_test.go

```diff
@@ -16,8 +16,15 @@ func TestQuesmaConfigurationLoading(t *testing.T) {
 
 	logLevelPassedAsEnvVar := "debug"
 	licenseKeyPassedAsEnvVar := "arbitraty-license-key"
-	os.Setenv("QUESMA_logging_level", logLevelPassedAsEnvVar) // overrides what's in the config file
-	os.Setenv("QUESMA_licenseKey", licenseKeyPassedAsEnvVar)  // overrides what's in the config file
+	os.Setenv("QUESMA_logging_level", logLevelPassedAsEnvVar)   // overrides what's in the config file
+	os.Setenv("QUESMA_licenseKey", licenseKeyPassedAsEnvVar)    // overrides what's in the config file
+	os.Setenv("QUESMA_backendConnectors_1_config_user", "user") // overrides what's in the config file
```

Comment:
It'd be cool if in addition to addressing backend connectors by index (`1`) you could address them by name, e.g. `QUESMA_backendConnectors_my-minimal-elasticsearch_config_user` (name from `name` field of connector)

But this probably will be messy if the name contains `_` separator.

Reply:
From that I'd expect:
```yaml
backendConnectors:
  my-minimal-elasticsearch:
    config:
      user: 
```
This goes against the syntax and introduces unnecessary ambiguity IMHO.

Reply:
Ok, sounds a valid use case, though don't know how to implement that easily.

Reply:
So while imperfect, I think it's easier to understand the final result 😉

Reply:
OK, I see. If there was an `backendConnectors` array already, we could deduce that `my-minimal-elasticsearch` is an "index" into that array, but if the configuration doesn't have `backendConnectors` at all we can't deduce that `backendConnectors` is an array.

I suggested this other approach since index into an array will get invalidated if the user adds another backend connector into the config - using a name would be more stable.

Reply:
@mieciu @avelanarius I added support by name and this looks much better.